lib/influxdb/writer/async.rb

```diff
@@ -0,0 +1,115 @@
+require 'thread'
+require "net/http"
+require "uri"
+
+module InfluxDB
+  module Writer # :nodoc: all
+    class Async
+      attr_reader :config, :client
+
+      def initialize(client, config)
+        @client = client
+        @config = config
+      end
+
+      def write(data)
+        data = data.is_a?(Array) ? data : [data]
```

Comment:
`Array(data)` does exactly this. http://ruby-doc.org/core-2.0.0/Kernel.html#method-i-Array